### DISCUSSION TITLE
How to get message response from a thread without calling ListMessages?

### DISCUSSION THREAD

**Discussion by discusser231** [betalgo/openai#654] [Q&A]
Hello,

I cannot find any information about this, hence why I am posting. I have run into an issue where I am using an assistant and sending a message to the thread. All works fine except for the fact that I wanted a way to get the specific message response

Here is a short snippet cut into what's needed for this context.

   ```cs
                var message = await OpenAIService.CreateMessage(_threadId, new MessageCreateRequest() {
                    Content = new MessageContentOneOfType(json)
                });

                if(message.RunId != null) {
                    await WaitForRunToFinishAsync(_threadId, message.RunId);
                }

                var threadRun = await OpenAIService.Beta.Runs.RunCreate(_threadId, new RunCreateRequest() {
                    AssistantId = _assistantId,
                });

                await WaitForRunToFinishAsync(_threadId, threadRun.Id);

                var messages = await OpenAIService.Beta.Messages.ListMessages(_threadId, new PaginationRequest() {
                    Limit = 1
                });

                var respondedMessage = messages.Data.First(); //This is the most recent response from above message
    
```
 This is the current method I am using to get the message response. However, I find this all a bit too weird. Surely there's a way to get the response message id after the run? Looking at the run object on completion, there's no real data that points to any sort of message id. If anyone can help show me the "proper way" to do it, I'd much appreciate it.

**Comment by tikkamasalla** [comment]
I am dealing with this issue as well, and the lack of documentation has been frustrating to say the least.

**Comment by jmuller3** [comment]
there's any number of ways to filter ListMessages - https://platform.openai.com/docs/api-reference/messages/listMessages

- limit + order: 1 and asc/desc to get first or last
- run_id to restrict to a specific run
- before/after to get [limit] messages occurring [before/after] a known message
